drivers/si1145/include/si1145_internals.h

```diff
@@ -0,0 +1,177 @@
+#define SI1145_REG_PS2_DATA1         (0x29)
+#define SI1145_REG_PS3_DATA0         (0x2A)
+#define SI1145_REG_PS3_DATA1         (0x2B)
+#define SI1145_REG_UV_INDEX0         (0x2C)
+#define SI1145_REG_UV_INDEX1         (0x2D)
+#define SI1145_REG_PARAM_RD          (0x2E)
+#define SI1145_REG_CHIP_STAT         (0x30)
+/** @} */
+
+/**
+ * @name SI1145 response register error codes
+ * @{
+ */
+#define SI1145_NO_ERROR             (0x00)
+#define SI1145_INVALID_SETTING      (0x80)
+#define SI1145_PS1_ADC_OVERFLOW     (0x88)
+#define SI1145_PS2_ADC_OVERFLOW     (0x89)
+#define SI1145_PS3_ADC_OVERFLOW     (0x8A)
+#define SI1145_ALS_VIS_ADC_OVERFLOW (0x8C)
+#define SI1145_ALS_IR_ADC_OVERFLOW  (0x8D)
+#define SI1145_UV_ADC_OVERFLOW      (0x8E)
+/** @} */
+
+/**
+ * @name SI1145 parameters RAM (accessed indirectly using PARAM_QUERY or
+ * PARAM_SET commands).
+ * @{
+ */
+#define SI1145_I2C_ADDR              (0x00)
+#define SI1145_CHLIST                (0x01)
+#define SI1145_PS_LED12_SELECT       (0x02)
+#define SI1145_PS_LED3_SELECT        (0x03)
+#define SI1145_PS_ENCODING           (0x05)
+#define SI1145_ALS_ENCODING          (0x06)
+#define SI1145_PS1_ADC_MUX           (0x07)
+#define SI1145_PS2_ADC_MUX           (0x08)
+#define SI1145_PS3_ADC_MUX           (0x09)
+#define SI1145_PS_ADC_COUNTER        (0x0A)
+#define SI1145_PS_ADC_GAIN           (0x0B)
+#define SI1145_PS_ADC_MISC           (0x0C)
+#define SI1145_ALS_IR_ADC_MUX        (0x0E)
+#define SI1145_AUX_ADCMUX            (0x0F)
+#define SI1145_ALS_VIS_ADC_COUNTER   (0x10)
+#define SI1145_ALS_VIS_ADC_GAIN      (0x11)
+#define SI1145_ALS_VIS_ADC_MISC      (0x12)
+#define SI1145_LED_REC               (0x1C)
+#define SI1145_ALS_IR_ADC_COUNTER    (0x1D)
+#define SI1145_ALS_IR_ADC_GAIN       (0x1E)
+#define SI1145_ALS_IR_ADC_MISC       (0x1F)
+/** @} */
+
+/**
+ * @name SI1145 constants
+ * @{
+ */
+#define SI1145_ID                    (0x45)
+#define SI1145_STARTUP_TIME          (30000UL)  /* 25ms */
+#define SI1145_INIT_VALUE            (0x17)
+#define SI1145_EN_UV                 (1 << 7)
+#define SI1145_EN_AUX                (1 << 6)
+#define SI1145_EN_ALS_IR             (1 << 5)
+#define SI1145_EN_ALS_VIS            (1 << 4)
+#define SI1145_EN_PS3                (1 << 2)
+#define SI1145_EN_PS2                (1 << 1)
+#define SI1145_EN_PS1                (1)
+#define SI1145_PS3_ALIGN             (1 << 6)
+#define SI1145_PS2_ALIGN             (1 << 5)
+#define SI1145_PS1_ALIGN             (1 << 4)
+#define SI1145_ALS_IR_ALIGN          (1 << 5)
+#define SI1145_ALS_VIS_ALIGN         (1 << 4)
+#define SI1145_PS_RANGE              (1 << 5)
+#define SI1145_PS_ADC_MODE           (1 << 2)
+#define SI1145_VIS_RANGE             (1 << 5)
+#define SI1145_IR_RANGE              (1 << 5)
+#define SI1145_UCOEF0_DEFAULT        (0x7B)
+#define SI1145_UCOEF1_DEFAULT        (0x6B)
+#define SI1145_UCOEF2_DEFAULT        (0x01)
+#define SI1145_UCOEF3_DEFAULT        (0x00)
+/*#define SI1145_UCOEF0_DEFAULT        (0x29)
```

Comment:
yes, I found other implementations [here](https://github.com/adafruit/Adafruit_SI1145_Library) with different coefficients than the ones given in the datasheet. I wanted to try.

I guess you noticed it's WIP ;)